kafka_consumer/datadog_checks/kafka_consumer/data/conf.yaml.example

```diff
@@ -1,75 +1,103 @@
-  # the self discovery of consumer groups, topics and partitions) the check
-  # will collect at metrics for at most 200 partitions.
-
+  # https://docs.datadoghq.com/developers/write_agent_check/?tab=agentv5#collection-interval
+  # min_collection_interval: 90
 
 instances:
+  # NOTE:
+  #   Old Kafka consumers store their offsets in Zookeeper.
+  #   New Kafka consumers store their offsets in Kafka itself.
+  #   This check can fetch offsets from either location (or both).
+  #   If consumer offsets exist in both Zookeeper and Kafka, they will
+  #   both be submitted and can be differentiated using the `source` tag.
+  #
+  #   For each check instance, you can explicitly list the consumer
+  #   groups / topics / partitions for which you want to fetch offsets,
+  #   or you can set the param `monitor_unlisted_consumer_groups` to True
+  #   and the check will attempt to fetch all known offsets.
+  #
+  # Here is the decision tree the check will use to decide which offsets to fetch from where:
+  #   Zookeeper:
+  #     - If `zk_connect_str` is specified, fetch consumer offsets from Zookeeper:
+  #       - If `monitor_unlisted_consumer_groups` is True:
+  #           Fetch all consumer offsets stored in Zookeeper
+  #       - Else:
+  #           Fetch the specified `consumer_groups` from Zookeeper. It will constrain itself
+  #           to whatever granularity you specify in consumer_groups/topics/partitions
+  #   Kafka:
+  #     - If `zk_connect_str` is not specified or `kafka_consumer_offsets` is True, fetch consumer offsets from Kafka:
+  #       - If `monitor_unlisted_consumer_groups` is True and broker >= 0.10.2:
+  #           Fetch all consumer offsets stored in Kafka
+  #       - If not `monitor_unlisted_consumer_groups` or broker < 0.10.2:
+  #           Fetch the specified `consumer_groups` from Kafka. It will constrain itself
+  #           to whatever granularity you specify in consumer_groups/topics/partitions.
+  #
   # In a production environment, it's often useful to specify multiple
   # Kafka / Zookeper nodes for a single check instance. This way you
   # only generate a single check process, but if one host goes down,
   # KafkaClient / KazooClient will try contacting the next host.
   # Details: https://github.com/DataDog/dd-agent/issues/2943
-  #
-  # If you wish to only collect consumer offsets from kafka, because
-  # you're using the new style consumers, you can comment out all
-  # zk_* configuration elements below.
-  # Please note that unlisted consumer groups are not supported at
-  # the moment when zookeeper consumer offset collection is disabled.
   - kafka_connect_str:
       - localhost:9092
       - another_kafka_broker:9092
-    zk_connect_str:
-      - localhost:2181
-      - another_zookeeper:2181
-    # zk_iteration_ival: 1  # how many seconds between ZK consumer offset
-                            # collections. If kafka consumer offsets disabled
-                            # this has no effect.
```

Comment:
I removed this param as it will be natively supported at the `instance` level in v6 of the agent. Since it unnecessarily complicated the code + user experience when configuring the check, I proactively removed it.